### ISSUE TITLE
Error with Public API

### ISSUE THREAD

**Issue by xela1** [bigbadblunt/homeassistant-lightwave2#79]
I'm currently unable to get working with the non public API (it just keeps retrying), so I thought I'd try the public one, but I get the following error:

```
2022-05-16 10:56:54 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry myemail for lightwave2
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 335, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/lightwave2/__init__.py", line 65, in async_setup_entry
    await link.async_get_hierarchy()
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 601, in async_get_hierarchy
    await self.async_update_featureset_states()
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 688, in async_update_featureset_states
    req = await self._async_postrequest("features/read", body)
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 550, in _async_postrequest
    return await req.json()
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 1103, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/html', url=URL('https://publicapi.lightwaverf.com/v1/features/read')

```

This seems to be the culprit

`2022-05-16 11:03:14 DEBUG (MainThread) [lightwave2.lightwave2] async_postrequest: Received API response 500 ((b'Date', b'Mon, 16 May 2022 10:03:11 GMT'), (b'Content-Type', b'text/html'), (b'Content-Length', b'177'), (b'Connection', b'keep-alive'), (b'Server', b'nginx/1.20.0')) <html>
`

**Comment by bigbadblunt** [comment]
This is weird

Are the app and the web app working for you? Can you double check your password is correct?

Can you give me the full log for this one please?

**Comment by xela1** [comment]
So I turned on debug logging and off the public API and it connected. Turned debug logging back off and it won't connect again. I'm travelling for a bit today, but will try again this evenign

**Comment by robtryan** [comment]
The integration will not load for me, with a similar error:

  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 223, in async_get_hierarchy
    await self.async_update_featureset_states()
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 269, in async_update_featureset_states
    _LOGGER.warning("update_featureset_states: failed to read feature ({}), returned {}".format(x.features[y][0], value))
KeyError: <lightwave2.lightwave2.LWRFFeature object at 0x7f9f7c9dc0>

 Also getting a timeout error on the LW app.  Will send LW an email.

**Comment by bigdmo007** [comment]
Yep. Down for me too

2022-05-16 22:21:25 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry {{account name}} for lightwave2
  File "/config/custom_components/lightwave2/__init__.py", line 45, in async_setup_entry
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 584, in async_get_hierarchy
  File "/usr/local/lib/python3.9/site-packages/lightwave2/lightwave2.py", line 633, in async_update_featureset_states
2022-05-16 22:23:30 DEBUG (MainThread) [custom_components.lightwave2.config_flow] Creating options form using existing options: {'lightwave_publicapi': True, 'lightwave_debug': True} 
2022-05-16 22:23:35 DEBUG (MainThread) [custom_components.lightwave2.config_flow] Received user input: {'lightwave_publicapi': False, 'lightwave_debug': True} 
  File "/config/custom_components/lightwave2/__init__.py", line 110, in reload_lw
  File "/config/custom_components/lightwave2/__init__.py", line 98, in async_remove_entry

Public API initially off in my setup. LW app works fine

**Comment by bigbadblunt** [comment]
I cannot reproduce this. Both public and non-public methods connect fine, both with and without logging enabled.

**Comment by bigdmo007** [comment]
I deleted and added twice before getting it back working.

Fault appeared limited to L22MK2 units also renamed in HA. 

Strange

**Comment by xela1** [comment]
> I cannot reproduce this. Both public and non-public methods connect fine, both with and without logging enabled.

I’ll try and run some more tests Friday when I’m home

**Comment by robtryan** [comment]
For me, LW come back to me the next day and said that they could see that I had connected successfully.
I tested and the LW app was working OK and the HA component loaded successfully too with publicApi disabled. It loads to when enabled but it errors with "failed to setup".  I've left it switched off and use the private one used by the app. Hoping for a fix to #69 now. So gratefull for this component. Thank you.
